File: fizzbuzz.js
#!/usr/bin/env node
var from = process.argv[2], 
	to = process.argv[3],
	i, results = [];

for(i = from; i <= to; i++)
	results.push(fizzbuzz(i));

console.log(results.join(", "));

function fizzbuzz(n) {
	if(fizz(n) && buzz(n))
		return "Fizz Buzz";
	if (fizz(n))
		return "Fizz";
	if (buzz(n))
		return "Buzz";
	return n;
}

function fizz(n) {
	return n % 3 === 0;
}

function buzz(n) {
	return n % 5 === 0;
}

File: fizzbuzz.hs
#!/usr/bin/runhaskell
import Data.List
import System.Environment
import Control.Applicative

main = do [from, to] <- map read <$> getArgs
          putStrLn . intercalate ", " . map fizzbuzz $ [from..to]

fizzbuzz n | fizz n && buzz n = "Fizz Buzz"
           | fizz n           = "Fizz"
           | buzz n           = "Buzz"
           | otherwise        = show n
    where fizz n = n `mod` 3 == 0
          buzz n = n `mod` 5 == 0
